subworkflows/local/amp.nf

```diff
@@ -65,8 +74,38 @@ workflow AMP {
 
         AMP_HMMER_HMMSEARCH ( ch_in_for_amp_hmmsearch )
         ch_versions = ch_versions.mix(AMP_HMMER_HMMSEARCH.out.versions)
+        GUNZIP2 ( AMP_HMMER_HMMSEARCH.out.output )
+        ch_hmmout = GUNZIP2.out.gunzip
+                        .map {
+                            meta_id, meta_hmm ->
+                            def meta_hmmsearch = [:]
+                            meta_hmmsearch['id'] = meta_id['id']
+                            [meta_hmmsearch, meta_hmm]
+                        }
+
+        ch_ampcombi_input = ch_ampcombi_input.mix(ch_hmmout)
     }
 
+    //AMPCOMBI
+    ch_ampcombi_input_new = ch_ampcombi_input
+        .groupTuple()
+        .join(ch_faa_for_ampcombi )
+        .multiMap{
+            input: [ it[0], it[1] ]
+            faa: it[2]
+        }
+
+    AMPCOMBI( ch_ampcombi_input_new.input, ch_ampcombi_input_new.faa , [])
+    ch_ampcombi_summaries = ch_ampcombi_summaries.mix(AMPCOMBI.out.csv)
+
+    //AMPCOMBI concatenation
+    ch_ampcombi_summaries_out = ch_ampcombi_summaries
+        .multiMap{
+                input: [ it[0] ]
+                summary: it[1]
+            }
+    ch_ampcombi_summaries_out.summary.collectFile(name: 'ampcombi_complete_summary.csv', storeDir: "${params.outdir}/reports/ampcombi", keepHeader:true)
```

Comment:
Nice!

Reply:
Does the keepHeader keep it for all files, or just the first?

Reply:
Just the first